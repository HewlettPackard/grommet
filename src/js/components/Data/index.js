export { Data } from './Data';
export { DataForm } from './DataForm';
