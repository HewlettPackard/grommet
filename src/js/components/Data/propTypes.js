import PropTypes from 'prop-types';

let PropType = {};
if (process.env.NODE_ENV !== 'production') {
  PropType = {
    data: PropTypes.arrayOf(PropTypes.shape({})),
  };
}
export const DataPropTypes = PropType;
