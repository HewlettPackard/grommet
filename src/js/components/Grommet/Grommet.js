import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
