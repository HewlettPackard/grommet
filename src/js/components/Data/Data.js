import React, { useEffect, useMemo, useState } from 'react';
import { Box } from '../Box';
import { DataFilters } from '../DataFilters';
import { DataSearch } from '../DataSearch';
import { DataSummary } from '../DataSummary';
import { Toolbar } from '../Toolbar';
import { DataContext } from '../../contexts/DataContext';
import { DataPropTypes } from './propTypes';
import { filter } from './filter';

const defaultView = {
  search: '',
  properties: {},
};

export const Data = ({
  children,
  data: dataProp,
  id = 'data',
  messages,
  onView,
  properties,
  toolbar,
  total,
  updateOn = 'submit',
  view: viewProp,
  ...rest
}) => {
  const [view, setView] = useState(viewProp);
  useEffect(() => setView(viewProp), [viewProp]);

  const data = useMemo(() => {
    if (onView) return dataProp;
    return filter(dataProp, view, properties);
  }, [dataProp, onView, properties, view]);

  // what we use for DataContext value
  const contextValue = useMemo(() => {
    const result = { id, messages, properties, updateOn, view };

    if (
      view?.search ||
      view?.sort ||
      (view?.properties && Object.keys(view.properties).length)
    ) {
      result.clearFilters = () => {
        const nextView = defaultView;
        setView(nextView);
        if (onView) onView(nextView);
      };
    }

    result.onView = (nextView) => {
      setView(nextView);
      if (onView) onView(nextView);
    };

    result.data = data;
    result.unfilteredData = dataProp;
    result.total = total !== undefined ? total : dataProp.length;

    return result;
  }, [data, dataProp, id, messages, onView, properties, total, updateOn, view]);

  let toolbarContent;
  if (toolbar) {
    toolbarContent = [
      <Toolbar key="toolbar">
        {(toolbar === true || toolbar === 'search') && <DataSearch />}
        {(toolbar === true || toolbar === 'filters') && <DataFilters drop />}
      </Toolbar>,
      <DataSummary key="summary" />,
    ];
  }

  return (
    <DataContext.Provider value={contextValue}>
      <Box id={id} flex={false} {...rest}>
        {toolbarContent}
        {children}
      </Box>
    </DataContext.Provider>
  );
};

Data.propTypes = DataPropTypes;
