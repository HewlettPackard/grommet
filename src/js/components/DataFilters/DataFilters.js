import React, { Children, useContext, useState } from 'react';
import { Filter } from 'grommet-icons/icons/Filter';
import { Box } from '../Box';
import { Button } from '../Button';
import { DataFilter } from '../DataFilter';
import { DataForm } from '../Data';
import { DropButton } from '../DropButton';
import { Header } from '../Header';
import { Heading } from '../Heading';
import { DataContext } from '../../contexts/DataContext';
import { MessageContext } from '../../contexts/MessageContext';
import { DataFiltersPropTypes } from './propTypes';

const dropProps = {
  align: { top: 'bottom', right: 'right' },
};

export const DataFilters = ({ drop, children, heading, ...rest }) => {
  const { clearFilters, data, messages, properties, view } =
    useContext(DataContext);
  const { format } = useContext(MessageContext);
  const [showContent, setShowContent] = useState();
  const controlled = drop;

  const clearControl = clearFilters && (
    <Box flex={false}>
      <Button
        label={format({
          id: 'dataFilters.clear',
          messages: messages?.dataFilters,
        })}
        onClick={clearFilters}
      />
    </Box>
  );

  let filters;
  if (Children.count(children) === 0) {
    let filtersFor;
    if (!properties && data && data.length) filtersFor = Object.keys(data[0]);
    else if (Array.isArray(properties)) filtersFor = properties;
    else if (typeof properties === 'object')
      filtersFor = Object.keys(properties);
    else filtersFor = [];
    filters = filtersFor.map((property) => (
      <DataFilter key={property} property={property} />
    ));
  }

  const content = (
    <DataForm
      pad={controlled ? 'medium' : undefined}
      gap="small"
      onDone={() => setShowContent(false)}
      {...(!controlled ? rest : {})}
    >
      {!drop && (
        <Header>
          <Heading margin="none" level={2} size="small">
            {heading ||
              format({
                id: 'dataFilters.heading',
                messages: messages?.dataFilters,
              })}
          </Heading>
          {!controlled && clearControl}
        </Header>
      )}
      {filters}
      {children}
    </DataForm>
  );

  if (!controlled) return content;

  let badge = 0;
  if (view?.properties) badge += Object.keys(view.properties).length;
  if (view?.search) badge += 1;
  if (view?.sort) badge += 1;
  if (!badge) badge = undefined;

  // drop
  const control = (
    <DropButton
      aria-label={format({
        id: 'dataFilters.open',
        messages: messages?.dataFilters,
      })}
      kind="toolbar"
      icon={<Filter />}
      dropProps={dropProps}
      dropContent={content}
      badge={badge}
      open={showContent}
      onOpen={() => setShowContent(undefined)}
      onClose={() => setShowContent(undefined)}
    />
  );

  return (
    <Box flex={false} direction="row" gap="small" {...rest}>
      {control}
      {clearControl}
    </Box>
  );
};

DataFilters.propTypes = DataFiltersPropTypes;
