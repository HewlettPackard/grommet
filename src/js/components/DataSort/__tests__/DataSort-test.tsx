import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Data } from '../../Data';
import { DataFilters } from '../../DataFilters';
import { Grommet } from '../../Grommet';
import { DataSort } from '..';

const data = [{ name: 'a' }, { name: 'b' }];

describe('DataSort', () => {
  test('renders', () => {
    const { container } = render(
      <Grommet>
        <Data data={data}>
          <DataFilters>
            <DataSort />
          </DataFilters>
        </Data>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
