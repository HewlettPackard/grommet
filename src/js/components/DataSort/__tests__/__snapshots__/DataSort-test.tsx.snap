// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSort drop 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c3 *[stroke*="#"],
.c3 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*="#"],
.c3 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus > circle,
.c2:focus > ellipse,
.c2:focus > line,
.c2:focus > path,
.c2:focus > polygon,
.c2:focus > polyline,
.c2:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c2:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) > circle,
.c2:focus:not(:focus-visible) > ellipse,
.c2:focus:not(:focus-visible) > line,
.c2:focus:not(:focus-visible) > path,
.c2:focus:not(:focus-visible) > polygon,
.c2:focus:not(:focus-visible) > polyline,
.c2:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div
    class="c1"
    id="test-data"
  >
    <button
      aria-label="Open sort"
      class="c2"
      id="test-data--sort-control"
      type="button"
    >
      <svg
        aria-label="Descend"
        class="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="m2 16 6 6 6-6M11 3h11M11 7h8m-8 4h5M8 22V2"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`DataSort drop 2`] = `
<button
  aria-label="Open sort"
  class="StyledButton-sc-323bzc-0 eHhrMZ"
  data-g-tabindex="none"
  id="test-data--sort-control"
  tabindex="-1"
  type="button"
>
  <svg
    aria-label="Descend"
    class="StyledIcon-sc-ofa7kd-0 jxjidg"
    viewBox="0 0 24 24"
  >
    <path
      d="m2 16 6 6 6-6M11 3h11M11 7h8m-8 4h5M8 22V2"
      fill="none"
      stroke="#000"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`DataSort drop 3`] = `
"@media only screen and (max-width: 768px) {
  .jkVdVC {
    margin-bottom: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .iPSxqg {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}
@media only screen and (max-width: 768px) {
  .knTyqE {
    margin-left: 6px;
    margin-right: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .fpBxMF {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}
@media only screen and (max-width: 768px) {
  .fpBxMF {
    padding: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .bTCisz {
    margin-right: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .hoZlGF {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}
@media only screen and (max-width: 768px) {
  .gsyDnL {
    margin-top: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .fNTNlK {
    height: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .gViPnC {
    width: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .bjyyAC {
    margin: 0px;
  }
}
@media only screen and (max-width: 768px) {
  .bjyyAC {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}"
`;

exports[`DataSort renders 1`] = `
.c17 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c17 g {
  fill: inherit;
  stroke: inherit;
}

.c17 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c17 *[stroke*="#"],
.c17 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c17 *[fill-rule],
.c17 *[FILL-RULE],
.c17 *[fill*="#"],
.c17 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: auto;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 24px;
  width: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c24 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c27 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c8 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c10:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible) > circle,
.c10:focus:not(:focus-visible) > ellipse,
.c10:focus:not(:focus-visible) > line,
.c10:focus:not(:focus-visible) > path,
.c10:focus:not(:focus-visible) > polygon,
.c10:focus:not(:focus-visible) > polyline,
.c10:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c26 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c26:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c26:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c26:focus > circle,
.c26:focus > ellipse,
.c26:focus > line,
.c26:focus > path,
.c26:focus > polygon,
.c26:focus > polyline,
.c26:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c26:focus::-moz-focus-inner {
  border: 0;
}

.c26:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c26:focus:not(:focus-visible) > circle,
.c26:focus:not(:focus-visible) > ellipse,
.c26:focus:not(:focus-visible) > line,
.c26:focus:not(:focus-visible) > path,
.c26:focus:not(:focus-visible) > polygon,
.c26:focus:not(:focus-visible) > polyline,
.c26:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c26:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c28 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c28:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c28:focus > circle,
.c28:focus > ellipse,
.c28:focus > line,
.c28:focus > path,
.c28:focus > polygon,
.c28:focus > polyline,
.c28:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c28:focus::-moz-focus-inner {
  border: 0;
}

.c28:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c28:focus:not(:focus-visible) > circle,
.c28:focus:not(:focus-visible) > ellipse,
.c28:focus:not(:focus-visible) > line,
.c28:focus:not(:focus-visible) > path,
.c28:focus:not(:focus-visible) > polygon,
.c28:focus:not(:focus-visible) > polyline,
.c28:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c28:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c20:hover input:not([disabled]) + div,
.c20:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c22 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c14 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c14::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c14::-moz-placeholder {
  color: #AAAAAA;
}

.c14:-ms-input-placeholder {
  color: #AAAAAA;
}

.c14::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c14::-moz-focus-inner {
  border: none;
  outline: none;
}

.c14:-moz-placeholder,
.c14::-moz-placeholder {
  opacity: 1;
}

.c13 {
  position: relative;
  width: 100%;
}

.c15 {
  cursor: pointer;
}

.c29 {
  opacity: 0;
}

.c2 {
  max-width: 100%;
}

.c6 {
  margin: 0px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c7 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c21 {
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c23 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:768px) {
  .c25 {
    margin-top: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c24 {
    height: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c27 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    font-size: 22px;
    line-height: 28px;
    max-width: 528px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="data"
  >
    <form
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <header
            class="c5"
          >
            <h2
              class="c6"
            >
              Filters
            </h2>
          </header>
          <div
            class="c7 "
          >
            <label
              class="c8"
              for="data--sort-property"
            >
              Sort by
            </label>
            <div
              class="c9 FormField__FormFieldContentBox-sc-m9hood-1"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Open Drop"
                class="c10 "
                id="data--sort-property"
                type="button"
              >
                <div
                  class="c11"
                >
                  <div
                    class="c12"
                  >
                    <div
                      class="c13"
                    >
                      <input
                        autocomplete="off"
                        class="c14 c15"
                        id="data--sort-property__input"
                        name="_sort.property"
                        readonly=""
                        tabindex="-1"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c16"
                    style="min-width: auto;"
                  >
                    <svg
                      aria-label="FormDown"
                      class="c17"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m18 9-6 6-6-6"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <div
            class="c7 "
          >
            <label
              class="c8"
              for="data--sort-direction"
            >
              Sort direction
            </label>
            <div
              class="c18 FormField__FormFieldContentBox-sc-m9hood-1"
            >
              <div
                class="c19"
                id="data--sort-direction"
                role="radiogroup"
              >
                <label
                  class="c20"
                >
                  <div
                    class="c21 "
                  >
                    <input
                      class="c22"
                      name="_sort.direction"
                      tabindex="0"
                      type="radio"
                      value="asc"
                    />
                    <div
                      class="c23 "
                    />
                  </div>
                  <span
                    class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
                  >
                    Ascending
                  </span>
                </label>
                <div
                  class="c24"
                />
                <label
                  class="c20"
                >
                  <div
                    class="c21 "
                  >
                    <input
                      class="c22"
                      name="_sort.direction"
                      tabindex="-1"
                      type="radio"
                      value="desc"
                    />
                    <div
                      class="c23 "
                    />
                  </div>
                  <span
                    class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
                  >
                    Descending
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
        <footer
          class="c25"
        >
          <button
            class="c26"
            type="submit"
          >
            Apply filters
          </button>
          <div
            class="c27"
          />
          <button
            aria-hidden="true"
            class="c28 c29"
            disabled=""
            tabindex="-1"
            type="reset"
          >
            Undo changes
          </button>
        </footer>
      </div>
    </form>
  </div>
</div>
`;
